WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackConnectionWebViewModel.swift

```diff
@@ -27,7 +27,14 @@ final class JetpackConnectionWebViewModel: AuthenticatedWebViewModel {
     }
 
     func handleRedirect(for url: URL?) {
-        // No-op
+        guard let path = url?.absoluteString else {
+            return
+        }
+        // When the web view navigates to the site address or Jetpack plans page,
+        // we can assume that the setup has completed.
+        if path.hasPrefix(siteURL) || path.hasPrefix(Constants.plansPage) {
```

Comment:
Nit: Would it make sense to create a helper method to handle the URL check? (And use that from here and in `func decidePolicy(for navigationURL: URL)`)